commit 68e25103c004d01e7e937ef0a183ac261a736a8f (refs/changes/49/849/3)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2016-11-07T09:41:25+00:00 (2 years, 11 months ago)
    
    TOOLS-1601 `sdcadm insts` error when a CN is deleted
    Reviewed by: Trent Mick <trent.mick@joyent.com>

diff --git a/lib/sdcadm.js b/lib/sdcadm.js
index 4e3e4e5..8849c11 100644
--- a/lib/sdcadm.js
+++ b/lib/sdcadm.js
@@ -715,10 +715,15 @@ SdcAdm.prototype.listInsts = function listInsts(opts, cb) {
                 // 'server_uuid'.
                 if (vm.server_uuid) {
                     inst.server = vm.server_uuid;
-                    inst.hostname = ctx.serverFromUuid[
-                        vm.server_uuid].hostname;
-                    inst.server_ip = ctx.serverAdminIpFromUuid[
-                        vm.server_uuid];
+                    if (ctx.serverFromUuid[vm.server_uuid]) {
+                        inst.hostname = ctx.serverFromUuid[
+                            vm.server_uuid].hostname;
+                    }
+
+                    if (ctx.serverAdminIpFromUuid) {
+                        inst.server_ip = ctx.serverAdminIpFromUuid[
+                            vm.server_uuid];
+                    }
                 }
 
                 var adminIp = vm.nics.filter(function (nic) {
